prusti-encoder/src/encoders/mir_impure.rs

```diff
@@ -41,14 +41,18 @@ const ENCODE_REACH_BB: bool = false;
 
 impl TaskEncoder for MirImpureEncoder {
     // TODO: local def id (+ promoted, substs, etc)
-    type TaskDescription<'vir> = DefId;
+    type TaskDescription<'vir> = (
+        DefId, // ID of the function
+        ty::GenericArgsRef<'vir>, // ? this should be the "signature", after applying the env/substs
+        Option<DefId>, // ID of the caller function, if any
```

Comment:
Why does this matter for impure methods? Are we generating monomorphised methods (for now)?

Reply:
Yep